### PR TITLE
hailctl batch

### PR DESCRIPTION
**Pull request by johnc1231**
First version of hailctl batch. I'm not a big batch user at this time so feedback on both code and desired functionality is welcome. @konradjk

### CODE REVIEW

File: hail/python/hailtop/hailctl/batch/delete.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from .batch_cli_utils import get_batch_if_exists
+
+def init_parser(parser):
+    parser.add_argument('id', type=int)
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/batch/delete.py]: help/description?

File: hail/python/hailtop/hailctl/batch/get.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import json
+
+import yaml
+
+from .batch_cli_utils import get_batch_if_exists
+
+def init_parser(parser):
+    parser.add_argument('batch_id', type=int)
+    parser.add_argument('-o', type=str, default='yaml')
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/batch/get.py]: help/descriptions for these. Also, you can defined possible values for `-o` - right now if I pass "yml" by accident I'll get a type error related to calling a None.

File: hail/python/hailtop/hailctl/batch/list_batches.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        except:
+            raise argparse.ArgumentTypeError("Boolean value expected for complete")
+
+    attributes = {}
+    if args.attributes:
+        for att in args.attributes:
+            att = att.strip()
+            key_value = att.split('=')
+            if len(key_value) != 2:
+                raise argparse.ArgumentTypeError('Attribute "{}" should contain exactly one equal sign')
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/batch/list_batches.py]: missing formatter.

I generally like f-strings here--it's very easy to do
```
f'Attribute {att!r} should contain exactly...'
```

The `!r` prints the repr, which will add its own quote.

Comment (tpoterba) [reply hail/python/hailtop/hailctl/batch/list_batches.py]: (and properly escape non-printable)

File: hail/python/hailtop/hailctl/batch/log.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+from .batch_cli_utils import get_job_if_exists
+
+def init_parser(parser):
+    parser.add_argument('batch_id', type=int)
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/batch/log.py]: help/description?

File: hail/python/hailtop/hailctl/batch/log.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    parser.add_argument('job_id', type=int)
+
+def main(args, pass_through_args, client):
+    maybe_job = get_job_if_exists(client, args.batch_id, args.job_id)
+    if maybe_job is None:
+        print("Job with id {} on batch {} not found".format(args.job_id, args.batch_id))
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/batch/log.py]: prefer f-strings

File: hail/python/hailtop/hailctl/batch/wait.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from .batch_cli_utils import get_batch_if_exists
+
+def init_parser(parser):
+    parser.add_argument('batch_id', type=int)
+
+def main(args, pass_through_args, client):
+    maybe_batch = get_batch_if_exists(client, args.batch_id)
+    if maybe_batch is None:
+        print("Batch with id {} not found".format(args.batch_id))
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/batch/wait.py]: prefer f-strings

Comment (tpoterba) [reply hail/python/hailtop/hailctl/batch/wait.py]: should this exit with a non-zero exit code? I dunno.

Comment (tpoterba) [reply hail/python/hailtop/hailctl/batch/wait.py]: or maybe raise an error?

File: hail/python/hailtop/hailctl/batch/list_batches.py (hunk 3 above)
Comment (johnc1231) [reply hail/python/hailtop/hailctl/batch/list_batches.py]: Nice didn't know that trick I've been doing `"".format(...)`

File: hail/python/hailtop/hailctl/batch/wait.py (hunk 6 above)
Comment (johnc1231) [reply hail/python/hailtop/hailctl/batch/wait.py]: I think nonzero exit code makes sense

Comment (johnc1231) [reply hail/python/hailtop/hailctl/batch/wait.py]: Just as an easy way for people using this in a script to check if things went wrong

File: hail/python/hailtop/hailctl/batch/log.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+from .batch_cli_utils import get_job_if_exists
+
```
Comment (jigold) [inline hail/python/hailtop/hailctl/batch/log.py]: FYI -- I'm in the middle of adding pod status #6489. Whomever goes in last should probably add that.

**Pull request merged**